backend/python/app/graphql/mutations/auth_mutation.py

```diff
@@ -42,3 +44,31 @@ def mutate(root, info, email):
         except Exception as e:
             error_message = getattr(e, "message", None)
             raise Exception(error_message if error_message else str(e))
+
+
+class Login(graphene.Mutation):
+    class Arguments:
+        email = graphene.String(required=True)
+        password = graphene.String(required=True)
+
+    access_token = graphene.String(required=True)
+    id = graphene.Int()
+    first_name = graphene.String(required=True)
+    last_name = graphene.String(required=True)
+    role = graphene.Field(RoleEnum, required=True)
+    email = graphene.String(required=True)
+
+    def mutate(root, info, email, password):
+        try:
+            auth_dto = auth_service.generate_token(email=email, password=password)
+            # TODO: is there a syntactically nicer way to destructure this?
```

Comment:
Something like the operator module looks interesting:
```
from operator import attrgetter
id, email, gender, username = attrgetter('id', 'email', 'gender', 'username')(current_user)
print(id, email, gender, username)
```
https://stackoverflow.com/questions/54785148/destructuring-dicts-and-objects-in-python

You could also add a deconstruct function to AuthDTO if you'd like (see first stack overflow response)